### PR TITLE
Remove Commons IO from Security CLI project

### PR DESCRIPTION
**Pull request by AthenaEryma**
The Security CLI project requires Commons IO, but only uses two methods from it which are fairly straightforward. This is a pretty heavy library to require for a couple methods, so we should avoid the burden of keeping it up to date by removing the dependency.

This PR is a relatively naive replacement of those methods and removal of Commons IO as a dependency from Security CLI.

Supercedes https://github.com/elastic/elasticsearch/pull/94414

### CODE REVIEW

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java [hunk 1]
```diff
@@ -942,14 +942,18 @@ private void notifyOfFailure(boolean inEnrollmentMode, Terminal terminal, Termin
...
     }
 
-    @SuppressForbidden(reason = "Commons IO lib uses the File API")
+    @SuppressForbidden(reason = "We need to interact with the file system in order to perform auto configuration")
     private void deleteDirectory(Path directory) throws IOException {
-        FileUtils.deleteDirectory(directory.toFile());
+        Files.walk(directory)
+            .sorted(Comparator.reverseOrder()) // Ensures directory contents are deleted first
+            .map(Path::toFile)
+            .map(File::delete);
```
Comment (tvernum) [inline x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:950]: I have no strong opinion, but while looking at this yesterday I noticed we also have:
- https://github.com/elastic/elasticsearch/blob/64798df2f344597767a675a71ba5ec6ddb7aa6ce/libs/core/src/main/java/org/elasticsearch/core/IOUtils.java#L196
and
- https://github.com/elastic/elasticsearch/blob/b2c9028384253416bd3e17de9e45af5eefe32764/server/src/main/java/org/elasticsearch/common/io/FileSystemUtils.java#L85

Perhaps one of them would do the job (or part of the job) here.

Comment (albertzaharovits) [reply x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:950]: ++ reusing `IOUtils#rm` sounds like a good idea.
I struggle to remember why I didn't do that in the first place.

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java [hunk 2]
```diff
@@ -942,14 +942,18 @@ private void notifyOfFailure(boolean inEnrollmentMode, Terminal terminal, Termin
...
     }
 
-    @SuppressForbidden(reason = "Commons IO lib uses the File API")
+    @SuppressForbidden(reason = "We need to interact with the file system in order to perform auto configuration")
     private void moveDirectory(Path srcDir, Path dstDir) throws IOException {
-        FileUtils.moveDirectory(srcDir.toFile(), dstDir.toFile());
+        Files.copy(srcDir, dstDir, StandardCopyOption.COPY_ATTRIBUTES);
+        deleteDirectory(srcDir);
```
Comment (albertzaharovits) [inline x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:956]: I looked at the code in `FileUtils#moveDirectory` and it first tries a `#rename`.
What do you think if we copy-paste the `FileUtils#moveDirectory` in the `IOUtils` class and use that here?

**Review by albertzaharovits — APPROVED** [review]

**Comment by albertzaharovits** [comment]
@tvernum kindly take another look. @gwbrown  and I agreed, I'll take it to address the last points.

File: x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java [hunk 3]
```diff
@@ -942,14 +943,18 @@ private void notifyOfFailure(boolean inEnrollmentMode, Terminal terminal, Termin
         }
     }
 
-    @SuppressForbidden(reason = "Commons IO lib uses the File API")
+    @SuppressForbidden(reason = "We need to interact with the file system in order to perform auto configuration")
```
Comment (tvernum) [inline x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java:946]: Is this still needed? I don't think `o.e.core.IOUtils` is forbidden

File: x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/AutoConfigureNodeTests.java [hunk 4]
```diff
@@ -291,8 +291,8 @@ private X509Certificate runAutoConfigAndReturnHTTPCertificate(Path configDir, Se
         return (X509Certificate) httpKeystore.getCertificate("http");
     }
 
-    @SuppressForbidden(reason = "Uses File API because the commons io library does, which is useful for file manipulation")
+    @SuppressForbidden(reason = "We need to interact with the file system in order to test auto configuration")
```
Comment (tvernum) [inline x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/AutoConfigureNodeTests.java:294]: As above

**Review by tvernum — APPROVED** [review]

**Comment by albertzaharovits** [comment]
@elasticmachine run elasticsearch-ci/part-2-fips

**Pull request merged**